c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1121,6 +1125,43 @@ void CachingAllocatorConfig::parseArgs(const char* env) {
   }
 }
 
+static std::string reportProcessMemoryInfo(int device) {
+#ifdef PYTORCH_C10_DRIVER_API_SUPPORTED
+  static c10::once_flag nvml_init;
+  c10::call_once(nvml_init, [] {
+    TORCH_INTERNAL_ASSERT(NVML_SUCCESS == DriverAPI::get()->nvmlInit_v2_());
+  });
+  nvmlDevice_t nvml_device;
+  TORCH_INTERNAL_ASSERT(
+      NVML_SUCCESS ==
+      DriverAPI::get()->nvmlDeviceGetHandleByIndex_v2_(device, &nvml_device));
```

Comment:
nvml index and runtime index are not necessarily the same (e.g. CUDA_VISIBLE_DEVICES changes runtime indices), python nvml calls go through annoying process of mapping one index to another. Should we just have a callback on python side when throwing OOMError?

Reply:
switched to using uuid which should be correct regardless of `CUDA_VISIBLE_DEVICES` and friends.

Reply:
Switched to PCI bus id so that we don't have to parse CUDA_VISIBLE_DEVICES and friends. Using PCI bus rather than uuid because the docs say uuid might need to initialize other GPUs to check their uuid, but pic bus id only has to initialize  the device it is specifying.